azurerm/resource_arm_automation_schedule.go

```diff
@@ -282,3 +364,90 @@ func resourceArmAutomationScheduleDelete(d *schema.ResourceData, meta interface{
 
 	return nil
 }
+
+func expandArmAutomationScheduleAdvanced(v []interface{}) (*automation.AdvancedSchedule, error) {
+
+	// Get the one and only advance schedule configuration
+	advancedSchedule := v[0].(map[string]interface{})
+	weekDays := advancedSchedule["week_days"].(*schema.Set).List()
+	monthDays := advancedSchedule["month_days"].(*schema.Set).List()
+	monthlyOccurrences := advancedSchedule["monthly_occurrence"].([]interface{})
+
+	expandedAdvancedSchedule := automation.AdvancedSchedule{}
+
+	if len(weekDays) > 0 {
```

Comment:
can we remove this if statement - since it means it's not possible to remove these values once they've been set.

Reply:
I've noticed an issue with this particular property. When creating the resource, if frequency is set to `Month` the `week_days` array cannot be set (even empty), otherwise the API returns an error. However during update it can and should be in order to reset it if it was removed from state. I've added a check for whether the current operation is a Create or Update and act accordingly. Perhaps separating the create and update methods would be a more clean solution?